packages/autocomplete-plugin-recent-searches/src/getTemplates.ts

```diff
@@ -11,33 +10,76 @@ export function getTemplates<TItem extends RecentSearchesItem>({
   onRemove,
 }: GetTemplatesParams): SourceTemplates<TItem> {
   return {
-    item({ item, root }) {
-      const content = document.createElement('div');
-      content.className = 'aa-ItemContent';
-      const icon = document.createElement('div');
-      icon.className = 'aa-ItemSourceIcon';
-      icon.innerHTML = recentIcon;
-      const title = document.createElement('div');
-      title.className = 'aa-ItemTitle';
-      title.innerHTML = reverseHighlightHit({
-        hit: item as any,
-        attribute: 'query',
+    item({ item, pragma, pragmaFrag }) {
+      return pragma(pragmaFrag, {
+        children: [
+          pragma('div', {
+            className: 'aa-ItemContent',
+            children: [
+              pragma('div', {
+                className: 'aa-ItemSourceIcon',
+                children: [
+                  pragma(
+                    'svg',
+                    {
+                      width: '20',
+                      height: '20',
+                      viewBox: '0 0 22 22',
+                      fill: 'currentColor',
+                    },
+                    pragma('path', {
+                      d: 'M0 0h24v24H0z',
+                      fill: 'none',
+                    }),
+                    pragma('path', {
+                      d:
+                        'M13 3c-4.97 0-9 4.03-9 9H1l3.89 3.89.07.14L9 12H6c0-3.87 3.13-7 7-7s7 3.13 7 7-3.13 7-7 7c-1.93 0-3.68-.79-4.94-2.06l-1.42 1.42C8.27 19.99 10.51 21 13 21c4.97 0 9-4.03 9-9s-4.03-9-9-9zm-1 5v5l4.28 2.54.72-1.21-3.5-2.08V8H12z',
+                    })
+                  ),
+                ],
```

Comment:
I feel like this should still be its own component, wdyt?

Reply:
We could but it'd need to use the provided `createElement`, so not too sure it's worth it to create a dedicated function. Wdyt?

Reply:
if we ever need to touch the file again, we can extract it in a function with the createElement being passed

Reply:
Yes, let's wait to see if we need the icon somewhere else.